### PR TITLE
UICIRC-97: Apply internationalization to all hardcoded strings

### PR DESCRIPTION
**Pull request by anastasiaHahanova**
## Purpose
Regarding [UICIRC-97](https://issues.folio.org/browse/UICIRC-97), all the hardcoded strings should be internationalized using the react-intl pattern constructs outlined [here](https://github.com/folio-org/stripes/blob/master/doc/i18n.md).

[React Intl docs](https://github.com/yahoo/react-intl/wiki)

### CODE REVIEW

File: settings/FixedDueDateScheduleDetail.js [hunk 1]
```diff
@@ -97,7 +96,7 @@ class FixedDueDateScheduleDetail extends React.Component {
           open={sections.generalInformation}
           id="generalInformation"
           onToggle={this.handleSectionToggle}
-          label="General information"
+          label={formatMessage({ id: 'ui-circulation.settings.fDDSform.about' })}
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleDetail.js]: Please use \<FormattedMessage\> here as it is now not wrapped in span and can be safely used.

File: settings/FixedDueDateScheduleDetail.js [hunk 2]
```diff
@@ -125,7 +124,7 @@ class FixedDueDateScheduleDetail extends React.Component {
           open={sections.fixedDueDateSchedule}
           id="fixedDueDateSchedule"
           onToggle={this.handleSectionToggle}
-          label="Schedule"
+          label={formatMessage({ id: 'ui-circulation.settings.fDDSform.schedule' })}
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleDetail.js]: Same, please use \<FormattedMessage\> component here

File: settings/FixedDueDateScheduleForm.js [hunk 3]
```diff
@@ -100,15 +105,14 @@ class FixedDueDateScheduleForm extends React.Component {
...
     return (
       <PaneMenu>
-        <button
-          id="clickable-close-fixedDueDateSchedule"
+        <IconButton
           onClick={onCancel}
+          icon="closeX"
+          size="medium"
+          iconClassName="closeIcon"
           title={formatMessage({ id: 'ui-circulation.settings.fDDSform.close' })}
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleForm.js]: Same, please use \<FormattedMessage\> component here

File: settings/FixedDueDateScheduleForm.js [hunk 4]
```diff
@@ -100,15 +105,14 @@ class FixedDueDateScheduleForm extends React.Component {
...
           aria-label={formatMessage({ id: 'ui-circulation.settings.fDDSform.closeLabel' })}
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleForm.js]: Same, please use \<FormattedMessage\> component here

File: settings/LoanPolicyDetail.js [hunk 5]
```diff
@@ -38,6 +38,12 @@ class LoanPolicyDetail extends React.Component {
     this.cViewMetaData = props.stripes.connect(ViewMetaData);
   }
 
+  translate(id) {
```
Comment (Godlevskyi) [inline settings/LoanPolicyDetail.js]: No need in this func as stated above Same,  \<FormattedMessage\> component could be used instead

File: settings/FixedDueDateScheduleDetail.js [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import _ from 'lodash';
-import { FormattedMessage, FormattedDate, intlShape, injectIntl } from 'react-intl';
+import { FormattedMessage, FormattedDate } from 'react-intl';
```
Comment (Godlevskyi) [inline settings/FixedDueDateScheduleDetail.js]: Please format each destructured variables on anew line

File: settings/StaffSlips/StaffSlipDetail.js [hunk 7]
```diff
@@ -55,32 +48,45 @@ class StaffSlipDetail extends React.Component {
     const { openDialog } = this.state;
     const { initialValues: staffSlip } = this.props;
     const contentComponent = this.parser.parseWithInstructions(staffSlip.template, () => true, this.rules);
+    const isActiveValue = staffSlip.active ? <FormattedMessage id="ui-circulation.settings.staffSlips.yes" /> :
+    <FormattedMessage id="ui-circulation.settings.staffSlips.no" />;
```
Comment (Godlevskyi) [inline settings/StaffSlips/StaffSlipDetail.js]: Please fix indentation

File: settings/StaffSlips/StaffSlipForm.js [hunk 8]
```diff
@@ -68,14 +61,15 @@ class StaffSlipForm extends React.Component {
   saveLastMenu() {
     const { pristine, submitting, initialValues } = this.props;
     const edit = initialValues && initialValues.id;
-    const saveLabel = edit ? this.translate('saveAndClose') : this.translate('createStaffSlip');
+    const saveLabel = edit ? <FormattedMessage id="ui-circulation.settings.staffSlips.saveAndClose" /> :
+    <FormattedMessage id="ui-circulation.settings.staffSlips.createStaffSlip" />;
```
Comment (Godlevskyi) [inline settings/StaffSlips/StaffSlipForm.js]: please fix indentation

File: settings/FixedDueDateScheduleForm.js [hunk 9]
```diff
@@ -156,30 +157,30 @@ class FixedDueDateScheduleForm extends React.Component {
...
-      intl: { formatMessage },
     } = this.props;
 
     if (selectedSet.id) {
       return (
         <div>
           <Icon size="small" icon="edit" />
-          <span>{`${formatMessage({ id: 'ui-circulation.settings.fDDS.edit' })}: ${selectedSet.name}`}</span>
+          <span>
+            <FormattedMessage id="ui-circulation.settings.fDDS.editLabel" values={{ name: selectedSet.name }} />
```
Comment (OleksandrYehorovEPAM) [inline settings/FixedDueDateScheduleForm.js]: Please write each property on a new line when there is more than one property

File: settings/FixedDueDateScheduleManager.js [hunk 10]
```diff
@@ -87,7 +93,8 @@ class FixedDueDateScheduleManager extends React.Component {
             const condA = (s1.from && s2.to) ? moment(s1.from).isBefore(s2.to) : false;
             const condB = (s1.to && s2.from) ? moment(s1.to).isAfter(s2.from) : false;
             if (condA && condB) {
-              errors.schedules = { _error: formatMessage({ id: 'ui-circulation.settings.fDDS.validate.overlappingDateRange' }, { num1: i + 1, num2: j + 1 }) };
+              const overlappingDateRangeMessage = <FormattedMessage id="ui-circulation.settings.fDDS.validate.overlappingDateRange" values={{ num1: i + 1, num2: j + 1 }} />;
```
Comment (OleksandrYehorovEPAM) [inline settings/FixedDueDateScheduleManager.js]: Please write each property on a new line when there is more than one property

**Pull request merged**